201-vm-linux-dynamic-data-disks/azuredeploy.json

```diff
@@ -288,7 +316,10 @@
           {
             "name": "[variables('networkSettings').subnet.dse.name]",
             "properties": {
-              "addressPrefix": "[variables('networkSettings').subnet.dse.prefix]"
+              "addressPrefix": "[variables('networkSettings').subnet.dse.prefix]",
+              "networkSecurityGroup": {
```

Comment:
wouldn't this be more secure on the nic?

Reply:
More secure in what way?  The NSG rules ultimately get applied the same way whether it's attached to the subnet or the NIC.

Our principal has been to broaden the security posture of these templates - if the NSG is attached to the subnet, then any other VM created later within the same subnet will also be behind the NSG by default.  But if the NSG were attached to the NIC, it would only protect that one VM and not any future VMs.